tests/examples.rs

```diff
@@ -0,0 +1,51 @@
+use std::ffi::OsStr;
+use std::fs;
+use std::process::{Command, Output};
+
+fn run_example<S: AsRef<str>>(name: S, args: &[&str]) -> Output {
+    let mut all_args = vec![
+        "run",
+        "--example",
+        name.as_ref(),
+        "--features",
+        "yaml unstable",
+        "--",
+    ];
+    all_args.extend_from_slice(args);
+
+    Command::new(env!("CARGO"))
+        .args(all_args)
+        .output()
+        .expect("failed to run example")
+}
+
+#[test]
+fn examples_are_functional() {
+    let example_paths = fs::read_dir("examples")
+        .expect("couldn't read examples directory")
+        .map(|result| result.expect("couldn't get directory entry").path())
+        .filter(|path| path.is_file() && path.extension().and_then(OsStr::to_str) == Some("rs"));
+
+    let mut example_count = 0;
+    for path in example_paths {
+        example_count += 1;
+
+        let example_name = path
+            .file_stem()
+            .and_then(OsStr::to_str)
+            .expect("unable to determine example name");
+
+        let help_output = run_example(example_name, &["--help"]);
+        assert!(
+            help_output.status.success(),
+            "{} --help exited with nonzero",
+            example_name,
+        );
+        assert!(
+            !help_output.stdout.is_empty(),
+            "{} --help had no output",
+            example_name,
+        );
```

Comment:
Some ideas for other assertions:

- Check for success and nonempty stdout when given `-h`/`--version`/`-V`.
- When given garbage CLI arguments, should exit with nonzero and emit something to stderr.

Please discuss!

Reply:
For example: https://github.com/mkantor/clap/commit/2187eff5106c04b2c3804c3f4bc8b1bcd477b286

I those additional assertions seem useful I'll add that commit to this branch. And please let me know if there's anything else I could/should check.